### PR TITLE
Fix tile bug & add more tests

### PR DESCRIPTION
**Pull request by eugene123tw**
### Summary

Good news, OV model maintains its accuracy without too much drop (1-2% on inst-seg). Interestingly, it even outperforms PyTorch model in terms of accuracy on detection models. Probably will look into it in another PR, but I think the risk for Geti is low at the moment. 

OV Model | dataset | Image Size | Tile Size | Version | Torch Test/mAP50 | OV Test/mAP_50 | Inference Duration(s)
-- | -- | -- | -- | -- | -- | -- | --
MaskRCNN-Eff | Vitens Aeromonas | 3500 x 3500 | 1663 x 1663 | 2.x (PR) | 0.9168 | 0.9061 |  59.016 
MaskRCNN-R50 | Vitens Aeromonas | 3500 x 3500 | 1663 x 1663 | 2.x (PR) | 0.8979 | 0.9052 |  72.322
MaskRCNN-Eff | Vitens Coliform | 3500 x 3500 | 1663 x 1663 | 2.x (PR) | 0.5311 | 0.5134 | 48.878
MaskRCNN-R50 | Vitens Coliform | 3500 x 3500 | 1663 x 1663 | 2.x (PR) | 0.4854 | 0.5054 |  70.749
yolox_s | Vitens Aeromonas | 3500 x 3500 | 1663 x 1663 | 2.x (PR) | 0.6198 | 0.7948 | 37.497
yolox_s | Vitens Coliform | 3500 x 3500 | 1663 x 1663 | 2.x (PR) | 0.3721 | 0.42 | 23.833
yolox_tiny | Vitens Aeromonas | 3500 x 3500 | 1663 x 1663 | 2.x (PR) | 0.9354 | 0.9529 | 34.481
yolox_tiny | Vitens Coliform | 3500 x 3500 | 1663 x 1663 | 2.x (PR) | 0.5171 | 0.5943 | 21.262
yolox_l | Vitens Aeromonas | 3500 x 3500 | 1663 x 1663 | 2.x (PR) | 0.7258 | 0.7989 | 71.366
yolox_l | Vitens Coliform | 3500 x 3500 | 1663 x 1663 | 2.x (PR) | 0.43689 | 0.4206 | 51.628

</body></html></div>

### Checklist

- [ ] I have added unit tests to cover my changes.​
- [ ] I have added integration tests to cover my changes.​
- [ ] I have added e2e tests for validation.
- [ ] I have added the description of my changes into CHANGELOG in my target branch (e.g., [CHANGELOG](https://github.com/openvinotoolkit/training_extensions/blob/develop/CHANGELOG.md) in develop).​
- [ ] I have updated the documentation in my target branch accordingly (e.g., [documentation](https://github.com/openvinotoolkit/training_extensions/tree/develop/docs) in develop).
- [ ] I have [linked related issues](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

### License

- [ ] I submit _my code changes_ under the same [Apache License](https://github.com/openvinotoolkit/training_extensions/blob/develop/LICENSE) that covers the project.
      Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example below).

```python
# Copyright (C) 2023 Intel Corporation
# SPDX-License-Identifier: Apache-2.0
```

### CODE REVIEW

File: tests/integration/test_tiling.py [hunk 1]
```diff
@@ -84,6 +94,32 @@ def test_tile_transform(self):
...
+        tiler.tiles_overlap = rnd_tile_overlap
+
+        mocker.patch("otx.core.data.dataset.tile.OTXTileTransform.__init__", return_value=None)
+        tile_transform = OTXTileTransform()
+        tile_transform._tile_size = (rnd_tile_size, rnd_tile_size)
+        tile_transform._overlap = (rnd_tile_overlap, rnd_tile_overlap)
+
+        dm_rois = [xywh_to_x1y1x2y2(*roi) for roi in tile_transform._extract_rois(dm_image)]
+        # 0 index in tiler is the full image so we skip it
+        assert np.allclose(dm_rois, tiler._tile(np_image)[1:])
```
Comment (vinnamkim) [inline tests/integration/test_tiling.py]: This seems more suitable for the unit test.

File: tests/integration/test_tiling.py [hunk 2]
```diff
@@ -147,3 +183,60 @@ def test_val_dataloader(self, fxt_det_data_config) -> None:
...
+            engine.test(exported_model_path, accelerator="cpu")
+
+            ov_model = OVInstanceSegmentationModel(model_name=exported_model_path, num_classes=3)
+
+            assert isinstance(
+                ov_model.model,
+                InstanceSegmentationTiler,
+            ), "Model should be an instance of InstanceSegmentationTiler"
+            assert engine.datamodule.config.tile_config.tile_size[0] == ov_model.model.tile_size
+            assert engine.datamodule.config.tile_config.overlap == ov_model.model.tiles_overlap
```
Comment (vinnamkim) [inline tests/integration/test_tiling.py]: Are these overlapped with the existing integration tests?

File: tests/integration/test_tiling.py [hunk 3]
```diff
@@ -18,7 +25,10 @@
 from otx.core.data.entity.detection import DetBatchDataEntity
 from otx.core.data.entity.tile import TileBatchDetDataEntity
 from otx.core.data.module import OTXDataModule
+from otx.core.model.entity.detection import OVDetectionModel
+from otx.core.model.entity.instance_segmentation import OVInstanceSegmentationModel
 from otx.core.types.task import OTXTaskType
+from otx.engine import Engine
 
 
 class TestOTXTiling:
```
Comment (harimkang) [inline tests/integration/test_tiling.py]: Similar to Vinnam's comment, wouldn't it be appropriate for us to move most of this to unit-test and for integration test, test the tiling recipe in the existing e2e (Do we need to have separate tests?)?

Comment (eugene123tw) [reply tests/integration/test_tiling.py]: CLI tests currently cover tile recipes, however, engine doesn't cover those recipes and I added a new test in engine API test.

File: tests/integration/test_tiling.py (hunk 1 above)
Comment (eugene123tw) [reply tests/integration/test_tiling.py]: I moved this under unit test folder.

File: tests/integration/test_tiling.py (hunk 2 above)
Comment (eugene123tw) [reply tests/integration/test_tiling.py]: CLI tests currently cover tile recipes, however, engine doesn't cover those recipes and I added a new test in engine API test.

**Review by harimkang — APPROVED** [review]

**Review by sungmanc — APPROVED** [review]

**Review by vinnamkim — APPROVED** [review]

**Pull request merged**